src/main/java/org/spongepowered/api/entity/player/Player.java

```diff
@@ -138,6 +137,38 @@
     void setGameMode(GameMode gameMode);
 
     /**
+     * Gets the location in dimension 0 of this player's bed spawn.
+     *
+     * @return The location of this player's bed spawn, or
+     * {@link Optional#absent} if it is not set
+     */
+    Optional<Vector3i> getBedPosition();
+
+    /**
+     * Sets the position in dimension 0 of this player's bed spawn. Passing
+     * {@link com.google.common.base.Optional#absent()} will clear it.
+     *
+     * @param location The new location of this player's bed spawn
+     */
+    void setBedPosition(Optional<Vector3i> location);
+
+    /**
+     * Returns whether this player is invisible to another given player.
+     *
+     * @param otherPlayer The player to toggle visibility of this player for
+     * @return Whether this player is invisible to another given player
+     */
+    boolean isInvisible(Player otherPlayer);
```

Comment:
I was more concerned about implementation when I wrote this method, as I wasn't sure whether it could be very easily implemented, but looking at it again I don't think it would be difficult to implement it in Living instead.